### DISCUSSION TITLE
What is the difference between "steps", "clock cycles" and "rows"?

### DISCUSSION THREAD

**Discussion by aborg-dev** [0xPolygonHermez/pilcom#64] [Q&A] [answered]
I've seen these three different words seemingly referring to the same concept of the length of the execution trace: [Steps](https://github.com/0xPolygonHermez/zkevm-proverjs/blob/07d92991778be82f66d8be55a3862d2843007221/src/main_executor.js#L44), [Clock cycles](https://github.com/0xPolygonHermez/zkevm-proverjs/blob/07d92991778be82f66d8be55a3862d2843007221/pil/arith.pil#L49), [Rows](https://github.com/0xPolygonHermez/zkevm-techdocs/blob/851cc4da42a57f108a6aed6b830d992d28c334c5/knowledge-layer/architecture/intro-proving-system/sections/prover-and-zkevm.tex#L12).

Is there any difference between them? Which one should we use to describe the time it takes to run an individual zkAsm program?

**Comment by zkronos73** [comment, marked as answer]
All are same, steps point of view a program, rows point of view trace, and clock because is like a processor, but really are same. I think for this case perhaps it's better steps, but all are the same.

**Reply by aborg-dev** [reply]
Ok, we will be using "steps" then aligning with proverjs.
Thank you!
